content/posts/2025-01-23-cli_whats_new_1.8.md

```diff
@@ -0,0 +1,229 @@
+
+```sh
+oc netobserv flows --background # Run a flow capture in the background
+
+oc netobserv follow             # Show current capture progression
+
+oc netobserv stop               # Stop the capture keeping the collector running
+
+oc netobserv copy               # Copy the output
+
+oc netobserv cleanup            # Remove all
+```
+
+### Customizable Namespace
+You can now customize the capture namespace using the `NETOBSERV_NAMESPACE` environment variable. When the CLI starts, it automatically checks if this namespace exists and will stop if it finds any conflict with a pending capture. This is particularly useful if you want to run captures in parallel.
+
+```sh
+NETOBSERV_NAMESPACE=my_ns oc netobserv [flows|packets|metrics|follow|stop|copy|cleanup]
+```
+
+### Subnets Labelling
+The tool can now read configurations from `cluster-config-v1` and `network` to identify **Machine**, **Pods**, and **Services** subnets using the `--get-subnets` option. This will automatically add `SrcSubnetLabel` and `DstSubnetLabel` to your flows.
+
+You will see subnets being configured during the creation of the agents:
+```sh
+creating flow-capture agents:
+opt: get_subnets, value: true
+Found subnets:
+    Services: "172.30.0.0/16"
+    Pods: "10.128.0.0/14"
+    Machines: "10.0.0.0/16"
+```
+
+![subnet labels]({page.image('cli-whats-new-1-8/subnets.png')})
+
+## Enhanced Data Filtering
+Filtering is crucial to gather precise network data without involving excessive resources and storage. The CLI focuses on this area, allowing you to deploy agents only where needed and fine-tune what's captured.
+
+### Assigning Agents to Nodes
+It's now possible to define agents `nodeSelector` to capture on a subset of nodes. You can rely on existing labels or create a dedicated one for this usage. For example, you can run:
+```sh
+oc netobserv flows --node-selector=kubernetes.io/hostname:my-node
+```
+to run the agents on the node with the `kubernetes.io/hostname:my-node` label.
+
+### eBPF Filters
+Agents recently introduced [the ability to filter](https://github.com/netobserv/netobserv-ebpf-agent/blob/main/docs/flow_filtering.md) on IPs, Ports, Protocol, Action, TCPFlags, and more simultaneously. You can now apply these filters in the CLI as shown below:
+
+```sh
+oc netobserv flows \              # Capture flows
+--protocol=TCP --port=8080 \      # either on TCP 8080
+or --protocol=UDP                 # or UDP
+```
+
+You will see filters being validated during the creation of the agents:
+```sh
+creating flow-capture agents:
+opt: filter_protocol, value: TCP
+opt: filter_port, value: 8080
+opt: add_filter
+opt: filter_protocol, value: UDP
+```
+
+![filters]({page.image('cli-whats-new-1-8/filters.png')})
+
+You can add as many filters as you want and separate them by or to create multiple capture scenarios.
```

Comment:
I guess it's not as many as you want, in the agent it is limited to 16, so I think the same limit exists here.
BTW - @msherif1234 - this limit of 16 seems to be documented on the [operator](https://github.com/netobserv/network-observability-operator/blob/main/apis/flowcollector/v1beta2/flowcollector_types.go#L303) side but not in the agent doc ( https://github.com/netobserv/netobserv-ebpf-agent/blob/main/docs/flow_filtering.md ), I guess we should add that

Reply:
```suggestion
You can add up to 16 sets of filters separated by `or`operator to create multiple capture scenarios.
```

Reply:
Would that work for you @jotak ? :point_up:

Reply:
yes except for the missing space after `or` :-)